packages/kit/src/runtime/server/endpoint.js

```diff
@@ -63,9 +63,9 @@ export default async function render_route(request, route) {
 			if (
 				typeof body === 'object' &&
 				!(body instanceof Uint8Array) &&
-				(!type || type === 'application/json')
+				(!type || type === 'application/json' || type === 'application/json; charset=utf-8')
```

Comment:
I would use `type.startsWith('application/json')` since the presence (or absence of the encoding does not change the behavior of the code